WalletWasabi.Tests/RegTests.cs

```diff
@@ -1010,7 +1010,65 @@ private void WalletTestsAsync_MemPoolService_TransactionReceived(object sender,
 				Assert.Contains("change of (outgoing, outgoing2)", wallet.Coins.Where(x => x.Height == bestHeight).Select(x => x.Label));
 				Assert.Contains("change of (outgoing, outgoing2)", keyManager.GetKeys().Select(x => x.Label));
 
-				#endregion Labeling
+				#endregion
+
+				#region AllowedInputsDisallowUnconfirmed
+
+				inputCountBefore = res.SpentCoins.Count();
+
+				receive = wallet.GetReceiveKey("AllowedInputsDisallowUnconfirmed").GetP2wpkhScript();
+
+				var allowedInputs = wallet.Coins.Where(x => !x.SpentOrLocked).Select(x => new TxoRef(x.TransactionId, x.Index)).Take(1);
+				var toSend = new[]{ new WalletService.Operation(receive, Money.Zero, "fizz")};
+
+				// covers:
+				// disallow unconfirmed with allowed inputs
+				// feeTarget < 2 // NOTE: need to correct alowing 0 and 1
+				res = await wallet.BuildTransactionAsync("password", toSend, 0, false, allowedInputs: allowedInputs);
+
+				activeOutput = res.InnerWalletOutputs.Single(x => x.ScriptPubKey == receive);
+				Assert.Single(res.InnerWalletOutputs);
+				Assert.Empty(res.OuterWalletOutputs);
+
+				Logger.LogInfo<RegTests>($"TxId: {res.Transaction.GetHash()}");
```

Comment:
Can you log these to Debug instead of Info, from time to time the test logs get too fat at testing.

Reply:
Not only these, I mean everything you logged here.

Reply:
Or even you could do Trace if you wish, up to you.